### PR TITLE
WIP Binary to geojson converter

### PR DESCRIPTION
**Pull request by kylebarron**
Right now this only parses single geometries.

Todo: 

- [ ] Parse MultiPolygons
- [x] Parse a collection of features and/or one feature with properties. Right now this only parses single geometries.
- [ ] Fix tests; add more test cases

For now I use the WKB test cases, which contains a mapping between GeoJSON and binary representations, but there are some edge cases that can be described in GeoJSON but not in our binary format. I.e. a geometry of type `MultiPoint` that only has one point. Those cases are assumed to be non-Multi. I'll need to copy some custom test cases for this into the `loaders.gl/gis` module.

### CODE REVIEW

File: modules/gis/src/index.js [hunk 1]
```diff
@@ -1 +1,2 @@
 export {geojsonToBinary, TEST_EXPORTS} from './lib/geojson-to-binary';
```
Comment (ibgreen) [inline modules/gis/src/index.js:1]: `geoJsonToBinary` (capital J) for consistency?

File: modules/gis/src/lib/binary-to-geojson.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
+export function binaryToGeoJson(data, type) {
```
Comment (ibgreen) [inline modules/gis/src/lib/binary-to-geojson.js:1]: Typescript file...

File: modules/gis/src/lib/binary-to-geojson.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+function parseGeometry(data, type) {
+  switch (type || parseType(data)) {
+    case 'point':
+      return pointToGeoJson(data);
+    case 'lineString':
+      return lineStringToGeoJson(data);
+    case 'polygon':
+      return polygonToGeoJson(data);
+    default:
+      throw new Error('Invalid type');
```
Comment (ibgreen) [inline modules/gis/src/lib/binary-to-geojson.js:29]: If we are going to have an error message let's have a slightly better error message as this will leave an end user non-plussed. At least "invalid geometry type". If not just assert with default message.

File: modules/gis/src/index.js (hunk 1 above)
Comment (kylebarron) [reply modules/gis/src/index.js:1]: Yes I was thinking about that... Not sure if an issue that it's already released

Comment (ibgreen) [reply modules/gis/src/index.js:1]: You can add always a deprecated section to avoid breaking old code.


```js
export {geoJsonToBinary} from '...

// DEPRECATED in v2.3
export {geoJsonToBinary as geojsonToBinary} from '...
```

You can also add `@deprecated` message to the typescript declaration comment.
